eb_hooks.py

```diff
@@ -3,21 +3,86 @@
 import os
 
 from easybuild.tools.build_log import EasyBuildError, print_msg
+from easybuild.tools.config import build_option, update_build_option
 from easybuild.tools.systemtools import POWER, get_cpu_architecture
 
 
+def get_eessi_envvar(eessi_envvar):
+    """Get an EESSI environment variable from the environment"""
+
+    eessi_envvar_value = os.getenv(eessi_envvar)
+    if eessi_envvar_value is None:
+        raise EasyBuildError("$%s is not defined!", eessi_envvar)
+
+    return eessi_envvar_value
+
+
+def get_rpath_override_dir(software_name):
+    # determine path to installations in software layer via $EESSI_SOFTWARE_PATH
+    eessi_software_path = get_eessi_envvar('EESSI_SOFTWARE_PATH')
+    eessi_pilot_version = get_eessi_envvar('EESSI_PILOT_VERSION')
+
+    # construct the rpath override directory
+    rpath_injection_dir = os.path.join(
+        # Make sure we are looking inside the `host_injections` directory
+        eessi_software_path.replace(eessi_pilot_version, os.path.join('host_injections', eessi_pilot_version), 1),
+        # Add the subdirectory for the specific software
+        'rpath_overrides',
+        software_name,
+        # We can't know the version, but this allows the use of a symlink to facilitate version upgrades without removing files
+        'system',
```

Comment:
This approach means that you can symlink to a host MPI installation directly (but you'd still need a module to pick up `bin`).